dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyQuiesceListener.java

```diff
@@ -0,0 +1,138 @@
+	
+	/**
+	 * Constructor
+	 */
+	public NettyQuiesceListener(NettyFrameworkImpl nettyBundle, ScheduledExecutorService scheduler, long timeout) {
+		this.bundle = nettyBundle;
+		this.scheduler = scheduler;
+		this.quiesceTimeout = (timeout > 0) ? timeout:0;
+	}
+			
+
+	/**
+	 * Runs the Quiesce verifier task at fixed intervals until the active channels have
+	 * been stopped or the Quiesce finish task finishes. The Quiesce finish task will 
+	 * be ran once after the Quiesce time has elapsed. 
+	 * @return
+	 */
+	public ScheduledFuture<?> startTasks() {
+		
+		this.quiesceFinishTask = scheduler.schedule( ()->{
+			if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            	Tr.debug(tc, "The quiesceFinishTask has started for " + this.bundle + ". Quiesce task will be scheduled to run for: " + this.bundle);
+            }
+			if(quiesceTask != null)
+				scheduler.schedule(quiesceTask, 0, TimeUnit.MILLISECONDS);
+			if(!this.quiesceVerifierTask.isDone()) {
+				if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	            	Tr.debug(tc, "The quiesceVerifierTask was found not to be done. Cancelling it...");
+	            }
+				this.quiesceVerifierTask.cancel(true);
+			}
+		}, quiesceTimeout, TimeUnit.MILLISECONDS);
+		
+		// TODO: Maybe we should check the map for the active channels if they have any connections left that could be better
+		
+		this.quiesceVerifierTask = scheduler.scheduleAtFixedRate(() -> {
+			if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            	Tr.debug(tc, "The quiesceVerifierTask has started,  checking endpoints...");
+            }
+			Set<Channel> activeChannels = this.bundle.getActiveChannels();
+			if(activeChannels.isEmpty()) {
+				if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	            	Tr.debug(tc, "No active connections have been found. Scheduling quiesce task and cancelling quiesce verifiers for: " + this.bundle);
+	            }
+				
+				if(quiesceTask != null)
+					scheduler.schedule(quiesceTask, 0, TimeUnit.MILLISECONDS);
+				
+				if(!this.quiesceFinishTask.isDone()){
+					if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+		            	Tr.debug(tc, "The quiesceFinishTask was found not to be done. Cancelling it...");
+		            }
+					this.quiesceFinishTask.cancel(true);
+				}
+				if(!this.quiesceVerifierTask.isDone()) {
+					if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+		            	Tr.debug(tc, "The quiesceVerifierTask was found not to be done. Cancelling it...");
+		            }
+					this.quiesceVerifierTask.cancel(true);
+				}			
+				
+			}else {
+				// Check if endpoint has open connections
+				if(TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	            	Tr.debug(tc, "Remaining endpoints still running. Verifying which can be closed...");
+	            }
+				// Iterate through active endpoints
+				for(Channel endpoint : activeChannels) {
```

Comment:
Perhaps activeChannels is something that we could change to activeEndpoints maybe that could simplify things since activeChannel no longer contain all the channels but the actual channel endpoints